src/js/structs/ServiceTree.js

```diff
@@ -115,16 +115,34 @@ module.exports = class ServiceTree extends Tree {
       }
 
       if (filter.health != null && filter.health.length !== 0) {
-        services = services.filter(function (service) {
-          return filter.health.some(function (healthValue) {
-            return service.getHealth().value === parseInt(healthValue, 10);
+        services = services.reduce(function (memo, service) {
+          filter.health.forEach(function (healthValue) {
+            if (service instanceof ServiceTree) {
+              memo = memo.concat(
+                service.filterItemsByFilter({health: [healthValue]}).getItems()
+              );
+              return;
+            }
+
+            if (service.getHealth().value === parseInt(healthValue, 10)) {
+              memo.push(service);
+            }
           });
-        });
+
+          return memo;
+        }, []);
```

Comment:
we need to do a recursive `reduce` because we want to traverse all of the children in order to build up all the services that match the filter.

previously, it would only get the top-most level in the filtering, and then for groups that contain children that match, it showed.

the current implementation does not show groups at all and shows only the children that match.